### DISCUSSION TITLE
What is different between the different models produced by the inverse folding script?

### DISCUSSION THREAD

**Discussion by avilella** [facebookresearch/esm#193] [Q&A]
This is a conceptual question rather than a bug report:

What is different between the different models produced by the inverse folding script?
If I request 100 different sequences outputted from running `esm/examples/inverse_folding/sample_sequences.py`, does it mean they are produced from the same model with a different seed?
If I wanted to take 1 forward from these 100, should I run `examples/inverse_folding/score_log_likelihoods.py` and use the one with the smallest negative value (the one closest to zero)?

**Comment by tomsercu** [comment]
1) That's right, same model but different random samples. The sampling method is multinomial sampling, and when temperature goes to zero this becomes greedy sampling.
2) Depending on your application there may be value in a set of diverse sequences or there may be other filtering criteria to select different candidates. But apart from that, yes the model thinks the largest probability sequence (smallest negative log likelihood)  is best.

**Comment by avilella** [comment]
I tried a few edits on proteins, by disrupting the aminoacids in places supposed to be important, and indeed, after feeding to `examples/inverse_folding/score_log_likelihoods.py`, those create more negative values in the log likelihood. To put some numbers in context, when making some small edits, it may change the second digit in the log-likelihood, e.g. from -1.12 to (-1.11 , -1.19), but when making radical changes, it can go down to -1.34 or -1.36.

**Reply by tomsercu** [reply]
That's great to hear!
